modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -607,3 +607,118 @@ func (suite *KeeperTestSuite) assertUpgradeError(actualError, expError error) {
 
 	suite.Require().True(errorsmod.IsOf(actualError, expError), actualError)
 }
+
+// TestAbortHandshake tests that when the channel handshake is aborted, the channel state
+// is restored the previous state and that an error receipt is written, and upgrade state which
+// is no longer required is deleted.
+func (suite *KeeperTestSuite) TestAbortHandshake() {
+	var (
+		path         *ibctesting.Path
+		upgradeError error
```

Comment:
note: this changed to `error` instead of `types.UpgradeError` because of [how nil checks on errors work](https://go.dev/doc/faq#nil_error) in go.